rise-storage.html

```diff
@@ -334,24 +349,96 @@
 
     /***************************************** STORAGE ******************************************/
 
+    _supportsLocalStorage: function() {
+      try {
+        return "localStorage" in window && window.localStorage !== null;
+      } catch (e) {
+        return false;
+      }
+     },
+
+    _getCachedSubscriptionStatus: function() {
+        return JSON.parse(sessionStorage.getItem(this.LOCAL_STORAGE_NAME+"-"+this.companyid));
+    },
+
+    _setCachedSubscriptionStatus: function (data) {
+      try {
+        sessionStorage.setItem(this.LOCAL_STORAGE_NAME+"-"+this.companyid, JSON.stringify(data));
+      } catch(e) {
+        console.warn(e.message);
+      }
+    },
+
+    _setSubscriptionStatus: function(status) {
+      if (this._supportsLocalStorage()) {
+        var now = new Date();
+        this._setCachedSubscriptionStatus({status: status, timestamp: now.getTime()});
+      } else {
+        this._isSubscribed = status;
+      }
+    },
+
+    _hasPassedTwentyFourHours: function (timestamp) {
+        var twentyFourInMilliseconds = 24 * 60 * 60 * 1000;
+        var now = new Date();
+        return now.getTime() > (timestamp + twentyFourInMilliseconds);
+    },
+
     _getStorageSubscription: function() {
+      if (this._supportsLocalStorage()) {
+        var subscriptionStatus = this._getCachedSubscriptionStatus();
 
+        if (!subscriptionStatus || this._hasPassedTwentyFourHours(subscriptionStatus.timestamp)) {
+          this._makeStorageSusbscriptionRequest();
+        } else {
+          if (subscriptionStatus.status) {
+            this._makeStorageRequest();
+          } else {
+            this._fireStorageSubscriptionError();
+          }
+        }
+      } else {
+        if (this._isLoading) {
+          this._makeStorageSusbscriptionRequest();
+        } else {
+          if (this._isSubscribed) {
+            this._makeStorageRequest();
+          } else {
+            this._fireStorageSubscriptionError();
```

Comment:
@stulees good point, we don't really need to fire it on a refresh. I will remove it. cheers